packages/big-design/src/components/Table/Search/Search.tsx

```diff
@@ -0,0 +1,36 @@
+import { SearchIcon } from '@bigcommerce/big-design-icons';
+import React, { FormEvent } from 'react';
+
+import { Button } from '../../Button';
+import { Flex, FlexItem } from '../../Flex';
+import { Form } from '../../Form';
+import { Input } from '../../Input';
+import { TableSearch } from '../types';
+
+export const Search: React.FC<TableSearch> = ({ value, onChange, onSubmit }) => {
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    onSubmit?.(event);
```

Comment:
Can we add a typeof guard here?
```suggestion
    if (typeof onSubmit === 'function') {
        onSubmit(event);
    }
```